### PR TITLE
Document Snaps transaction insights

### LINKED ISSUE #1260: Add Snaps transaction insights feature page
Add a topic under **Snaps > Features** for transaction insights. This page should describe how to use the `onTransaction` entry point and `endowment:transaction-insight` permission to provide transaction insights. It can take inspiration from and link to the [Transaction insights tutorial](https://docs.metamask.io/snaps/learn/tutorials/transaction-insights/). It should also link to the [transaction insights example package](https://github.com/MetaMask/snaps/tree/main/packages/examples/packages/transaction-insights).

This topic should also cover the transaction severity level feature.

The transaction insights card on the Snaps intro page can then link to this feature page (maybe remove txn severity level card).

### PR DESCRIPTION
**Pull request by alexandratran**
# Description

Add feature page for transaction insights. Also reorder feature cards on index page in alphabetical order, and make minor changes to related content for consistency and simplicity.

## Issue(s) fixed

Fixes #1260 

## Preview

https://docs.metamask.io/1260-txn-insights/snaps/features/transaction-insights/

## Checklist

Complete this checklist before merging your PR:

- [x] If this PR contains a major change to the documentation content, I have added an entry to the top of the ["What's new?"](https://github.com/MetaMask/metamask-docs/blob/main/docs/whats-new.md) page.
- [ ] The proposed changes have been reviewed and approved by a member of the documentation team.

### CODE REVIEW

File: snaps/learn/tutorials/gas-estimation.md [hunk 1]
```diff
@@ -128,9 +128,9 @@ Edit the `files` array and add the `images/` folder:
 
 ### 3. Enable network access
 
-To enable your Snap to use the `fetch` API, enable the
-[`endowment:network-access`](../../reference/permissions.md#endowmentnetwork-access) 
-permission by adding `"endowment:network-access": {}` to the `initialPermissions` object in `snap.manifest.json`:
+To enable your Snap to [access the internet using the `fetch` API](../../features/network-access.md),
+request the [`endowment:network-access`](../../reference/permissions.md#endowmentnetwork-access) 
+permission in `packages/snap/snap.manifest.json`:
```
Comment (Mrtenz) [inline snaps/learn/tutorials/gas-estimation.md:133]: Not sure about this change. `packages/snap` assumes the template Snap, which may not always be the case. I feel like it's better to just mention `snap.manifest.json`.

Comment (alexandratran) [reply snaps/learn/tutorials/gas-estimation.md:133]: This change is specifically for [this tutorial](https://docs.metamask.io/snaps/learn/tutorials/gas-estimation/), which uses the template generated by `create-snap`, so being specific with the file location should be fine (and consistent with the rest of the tutorial).

Comment (Mrtenz) [reply snaps/learn/tutorials/gas-estimation.md:133]: Ah, right. Looks good to me in that case!

**Review by Mrtenz — APPROVED** [review]

File: snaps/features/transaction-insights.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
+---
+description: Provide transaction insights in MetaMask's transaction confirmation window.
+toc_max_heading_level: 4
+sidebar_position: 13
+---
+
+# Transaction insights
+
+You can provide transaction insights in MetaMask's transaction confirmation window before a user
```
Comment (bgravenorst) [inline snaps/features/transaction-insights.md:9]: Perhaps add a high level overview about what transactions insights are, and why you may want to create one? Are there any limitations?

File: snaps/features/transaction-insights.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+signs a transaction.
+For example, you can show the user the percentage of gas fees they would pay for their transaction.
+
+## Steps
```
Comment (bgravenorst) [inline snaps/features/transaction-insights.md:13]: Are there any prerequisites?

**Review by bgravenorst — APPROVED** [review]
Some questions. Can be ignored if you feel it's irrelevant.

**Pull request merged**